### DISCUSSION TITLE
Brace Highlighting Functionality Error. Need Guidance

### DISCUSSION THREAD

**Discussion by TheMarvelFan** [apache/netbeans#6497] [General]
Hi. I recently started contributing to open source, and decided that I take my first issue as #5024 . I added the support for angle brackets in various places and files where I thought it was nacessary. To test my functionality I built the project using Ant, and upon running, there is some error in the code, or maybe I am missing something in the algorithm. When I type the following line in the editor, instead of highlighting the open and close angle brackets (like it does braces), it gives me a red highlight on only one of the angle brackets, which is the one i kept my caret on: 
`List<Integer> al = new ArrayList<>();` 
The following are images for reference:
![Screenshot 2023-09-28 140602](https://github.com/apache/netbeans/assets/93257324/cf75d667-53a5-49e2-8a6c-91daba598e0a)

The way it is working:
![Screenshot 2023-09-28 140642](https://github.com/apache/netbeans/assets/93257324/6726bf51-6a6d-4ef8-a838-f186a8d6f652)
The way it should work:
![Screenshot 2023-09-28 140843](https://github.com/apache/netbeans/assets/93257324/da2cd0bc-da70-4572-a4b1-78443ffedd3c)
(should have the same highlighting as the parentheses)
If you can help, please reply so we can talk about implementation details.

**Comment by Chris2011** [comment]
I guess the problem here is that those diamond characters are part of the generic impl of List and ArrayList. You didn't import the corresponding packages. Try to add the imports and you will see the correct color matching.

**Reply by TheMarvelFan** [reply]
I tried that. Still the same error.
![Screenshot 2023-09-29 191556](https://github.com/apache/netbeans/assets/93257324/e4d6bb92-5329-44e4-97a8-edd4cb75c832)

**Reply by Chris2011** [reply]
I just have no color for this and in general this color means that there is a bracket missing. Is your file runnable your project compilable?

**Reply by TheMarvelFan** [reply]
Yes its compatible and is showing some output on the terminal. I think the error is in the source code somewhere. I will look into it a bit more, but I am pretty sure I covered everything when I made changes.
